src/Database/Connectors/MySqlConnector.php

```diff
@@ -32,25 +32,27 @@ public function connect(array $config): \PDO
      */
     protected function getDsn(array $config): string
     {
-        extract($config, EXTR_SKIP);
-
-        // @TODO refactor this code as the use of extract() is a bad practice, prone to create bugs
+        // All these if, empty, are here to clean the legacy code before the fork. See the git history.
+        $host = array_key_exists('host', $config) ? $config['host'] : null;
+        $unix_socket = array_key_exists('unix_socket', $config) ? $config['unix_socket'] : null;
+        $port = array_key_exists('port', $config) ? $config['port'] : null;
+        $database = array_key_exists('database', $config) ? $config['database'] : null;
```

Comment:
Peut être un array_map pour rendre le code moins rébarbatif (ou une méthode pour prendre ça en charge) ?
```suggestion
        $normalized = array_map(
            function (string $key) use ($config) {
                return
                    array_key_exists($key, $config)
                    ? $config[$key]
                    : null
                ;
            },
            [
                'host',
                'unix_socket',
                'port',
                'database',
            ]
        );
```
(puis après on utilise l'array `$normalized`)

Reply:
je vais tester sur un pour voir.

Reply:
So I did some tests, but my answer would be "yes, but later" :-D

Because doing a validation would imply doing a nice one, not a basic one.
During my investigation I noticed that empty keys are sent because the original developer
always call them (with this way, its code never crashes).

So let's say that doing a clean validation would imply to go in every call and 
1. Validate the type (array, string, scalar...)
2. Be sure that no useless keys are set in the params.

I am going to open another ticket for that, and currently, we just keep the array_key exist, which has the merit to init the value and do not hide code to linters and developers, which was the original purpose of this ticket :)